docker/docker-compose.yml

```diff
@@ -10,93 +10,136 @@ etcd:
   command: >
     --listen-client-urls 'http://etcd:4001'
     --advertise-client-urls 'http://etcd:4001'
-#zookeeper:
-#  image: jplock/zookeeper:3.4.5
-#  ports: [ "2181:2181" ]
-mesos:
-  hostname: mesos
+    --initial-cluster-state new
+mesosmaster1:
+  hostname: mesosmaster1
   image: mesosphere/mesos:0.22.0-1.0.ubuntu1404
-  entrypoint: mesos-local
+  entrypoint: mesos-master
   ports: [ "5050:5050" ]
   environment:
-  - MESOS_HOSTNAME=mesos
+  - MESOS_HOSTNAME=mesosmaster1
   - MESOS_PORT=5050
   - MESOS_LOG_DIR=/var/log/mesos
   - MESOS_QUORUM=1
   - MESOS_REGISTRY=in_memory
   - MESOS_WORK_DIR=/var/lib/mesos
-#  - MESOS_ZK=zk://zookeeper:2181/mesos
   links:
   - etcd
   - "ambassador:apiserver"
+mesosslave1:
+  hostname: mesosslave1
+  privileged: true
+  image: mesosphere/kubernetes-mesos-slave:latest
+  ports: [ "5051:5051" ]
+  environment:
+  - MESOS_HOSTNAME=mesosslave1
```

Comment:
If you don't set the hostname at all, mesos-slave will use the internal eth0 IP 172.x.y.z for that. Mesos-master then will communicate directly using this IP. This will make the ambassador unnecessary.